### PR TITLE
Display commit status on landing page of repo

### PR DESCRIPTION
**Pull request by DblK**
This will implement a part of #996 and so #1029.

The modification of the ui will be like this:
image

Status is now displayed.

### CODE REVIEW

**Comment by DblK** [comment]
To follow the merged #1688, this could be done before adding the information for PR.

**Comment by appIeboy** [comment]
<p align=right><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup><sub><sup>#1798 @lunny @tboerger @sapk @cez81 @geek1011 @lafriks @ethantkoenig @andreynering @DblK </sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></sup></sub></p>

**Comment by lunny** [comment]
CI failed.

**Comment by DblK** [comment]
I saw, I need to rebase on master so the build will not failed since RepoCreationNum as been renamed in MaxCreationLimit.

File: routers/repo/view.go
Comment (ethantkoenig) [inline routers/repo/view.go]: Is there a reason you don't just use two local variables? The `SignCommit` field of `commit` never comes into play, so I think that just using variables would be simpler.

**Review by ethantkoenig — CHANGES_REQUESTED** [review]

File: routers/repo/view.go
Comment (DblK) [reply routers/repo/view.go]: Good point. I took some portion of code from the display. I will use local variable instead.

File: templates/repo/view_list.tmpl
Comment (lafriks) [inline templates/repo/view_list.tmpl]: Can this code be moved to separate template and reused in both places?

File: routers/repo/view.go
Comment (bkcsoft) [inline routers/repo/view.go]: ~~Why not get the CommitStatus for the latest SHA? (As-is now you're introducting race-conditions)~~

**Comment by bkcsoft** [comment]
@DblK Any update? 🙂

**Comment by DblK** [comment]
Kind of busy right now, I'll work on it later.

**Comment by appleboy** [comment]
@DblK Any updates on this?

**Comment by lunny** [comment]
@DblK if you are some busy, could some maintainers help you to finish this PR?

**Comment by DblK** [comment]
If some maintainers want to finish then go for it.

File: routers/repo/view.go
Comment (Morlinest) [reply routers/repo/view.go]: This is duplicated code from ParseCommitsWithStatus function.

File: routers/repo/view.go
Comment (lunny) [reply routers/repo/view.go]: done.

File: templates/repo/view_list.tmpl
Comment (lunny) [reply templates/repo/view_list.tmpl]: done.

File: routers/repo/view.go
Comment (lunny) [reply routers/repo/view.go]: Since `ctx.Repo.Commit` has parsed and not used ParseCommitsWithStatus, I will simple handle here and maybe another PR to do further improvement.

**Review by appleboy — APPROVED** [review]

**Comment by lunny** [comment]
See below three places status and all have links to CI now.

**Comment by appleboy** [comment]
LGTM

**Comment by lafriks** [comment]
LGTM

**Comment by lunny** [comment]
@lafriks done.

**Comment by lunny** [comment]
@ethantkoenig oh sorry, just found need your approval after I merged.

**Comment by ethantkoenig** [comment]
@lunny No worries, it looks my suggestion was addressed nonetheless

**Comment by lunny** [comment]
@ethantkoenig OK.

**Pull request merged**
